docs/tutorials/essentials/part-8-rtk-query-advanced.md

```diff
@@ -0,0 +1,1092 @@
+
+We could add a `'Post'` tag to both the `getPost` query and the `editPost` mutation, but that would force all the other individual posts to be refetched as well. Fortunately, **RTK Query lets us define specific tags, which let us be more selective in invalidating data**. These specific tags look like `{type: 'Post', id: 123}`.
+
+Our `getPosts` query defines a `providesTags` field that is an array of strings. The `providesTags` field can also accept a callback function that receives the `result` and `arg`, and returns an array. This allows us to create tag entries based on IDs of data that is being fetched. Similarly, `invalidatesTags` can be a callback as well.
+
+In order to get the right behavior, we need to set up each endpoint with the right tags:
+
+- `getPosts`: provides a general `'Post'` tag for the whole list, as well as a specific `{type: 'Post', id}` tag for each received post object
+- `getPost`: provides a specific `{type: 'Post', id}` object for the individual post object
+- `addNewPost`: invalidates the general `'Post'` tag, to refetch the whole list
+- `editPost`: invalidates the specific `{type: 'Post', id}` tag. This will force a refetch of both the _individual_ post from `getPost`, as well as the _entire_ list of posts from `getPosts`, because they both provide a tag that matches that `{type, id}` value.
+
+```js title="features/api/apiSlice.js"
+export const apiSlice = createApi({
+  reducerPath: 'api',
+  baseQuery: fetchBaseQuery({ baseUrl: '/fakeApi' }),
+  tagTypes: ['Post'],
+  endpoints: builder => ({
+    getPosts: builder.query({
+      query: () => '/posts',
+      // highlight-start
+      providesTags: (result = [], error, arg) => [
+        'Post',
+        ...result.map(({ id }) => ({ type: 'Post', id }))
+      ]
+      // highlight-end
+    }),
+    getPost: builder.query({
+      query: postId => `/posts/${postId}`,
+      // highlight-start
+      providesTags: (result, error, arg) => [{ type: 'Post', id: arg.id }]
+      // highlight-end
+    }),
+    addNewPost: builder.mutation({
+      query: initialPost => ({
+        url: '/posts',
+        method: 'POST',
+        body: initialPost
+      }),
+      // highlight-next-line
+      invalidatesTags: ['Post']
+    }),
+    editPost: builder.mutation({
+      query: post => ({
+        url: `posts/${post.id}`,
+        method: 'PATCH',
+        body: post
+      }),
+      // highlight-next-line
+      invalidatesTags: (result, error, arg) => [{ type: 'Post', id: arg.id }]
+    })
+  })
+})
+```
+
+It's possible for the `result` argument in these callbacks to be undefined if the response has no data or there's an error, so we have to handle that safely. For `getPosts` we can do that by using a default argument array value to map over, and for `getPost` we just return an empty array if `result` isn't there. For `editPost`, we know the ID of the post from the partial post object that was passed into the trigger function, so we can read it from there.
```

Comment:
> and for `getPost` we just return an empty array if `result` isn't there

I'm not sure of the best way to re-word it, but `getPost` now always provides tags, so this line should be altered to say that it uses the ID of the post from the argument